catalyst/core/runner.py

```diff
@@ -151,6 +151,49 @@ def _get_experiment_components(
 
         return model, criterion, optimizer, scheduler, device
 
+    def _process_callbacks_for_stage(self, callbacks):
+        loggers = utils.process_callbacks(
+            OrderedDict(
+                [
+                    (k, v) for k, v in callbacks.items()
+                    if isinstance(v, LoggerCallback)
+                ]
+            )
+        )
+        callbacks = utils.process_callbacks(
+            OrderedDict(
+                [
+                    (k, v) for k, v in callbacks.items()
+                    if not isinstance(v, LoggerCallback)
+                ]
+            )
+        )
+        self.state.loggers = loggers
+        self.loggers = loggers
+        self.callbacks = callbacks
+        # Prepare events dictionary
+        start_callbacks = OrderedDict(**loggers, **callbacks)
+        end_callbacks = OrderedDict(**callbacks, **loggers)
+        start_events = ["stage_start", "epoch_start", "batch_start", "loader_start"]
+        end_events = ["stage_end", "epoch_end", "batch_end", "loader_end", "exception"]
+        stage_callbacks = {}
+        for event in start_events:
+            fn_name = f"on_{event}"
+            filtered_callbacks = OrderedDict([
+                (k, v) for k, v in start_callbacks.items()
+                if not getattr(v.__class__, fn_name).__code__ is getattr(Callback, fn_name).__code__
```

Comment:
could you please 
- create new function like `check_if_fn_empty` (think about naming ;) )
- place it to utils.misc
- write some tests, to check, that is works correctly
?